lib/backend/syncersv1/updateprocessors/workloadendpointprocessor.go

```diff
@@ -0,0 +1,200 @@
+// Copyright (c) 2017 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package updateprocessors
+
+import (
+	"errors"
+	"net"
+
+	"github.com/projectcalico/libcalico-go/lib/apiv2"
+	"github.com/projectcalico/libcalico-go/lib/backend/model"
+	"github.com/projectcalico/libcalico-go/lib/backend/watchersyncer"
+	"github.com/projectcalico/libcalico-go/lib/names"
+	cnet "github.com/projectcalico/libcalico-go/lib/net"
+)
+
+// Create a new SyncerUpdateProcessor to sync WorkloadEndpoint data in v1 format for
+// consumption by Felix.
+func NewWorkloadEndpointUpdateProcessor() watchersyncer.SyncerUpdateProcessor {
+	return NewGeneralUpdateProcessor(apiv2.KindWorkloadEndpoint, convertWorkloadEndpointV2ToV1)
+}
+
+// Convert v2 KVPair to the equivalent v1 KVPair.
+func convertWorkloadEndpointV2ToV1(kvp *model.KVPair) (*model.KVPair, error) {
+	// Validate against incorrect key/value kinds.  This indicates a code bug rather
+	// than a user error.
+	v2key, ok := kvp.Key.(model.ResourceKey)
+	if !ok || v2key.Kind != apiv2.KindWorkloadEndpoint {
+		return nil, errors.New("Key is not a valid WorkloadEndpoint resource key")
+	}
+
+	if kvp.Value == nil {
+		return nil, errors.New("Deletion attempted without enough information to form a v1 WorkloadEndpoint key")
+	}
+
+	v2res, ok := kvp.Value.(*apiv2.WorkloadEndpoint)
+	if !ok {
+		return nil, errors.New("Value is not a valid WorkloadEndpoint resource value")
+	}
+
+	v1key, err := convertWorkloadEndpointV2ToV1Key(v2res)
+	if err != nil {
+		return nil, err
+	}
+
+	v1value, err := convertWorkloadEndpointV2ToV1Value(v2res)
+	if err != nil {
+		// Currently ignore the error so that incorrect values get skipped instead of erroring out
+		return &model.KVPair{
+			Key: v1key,
+		}, nil
+	}
+
+	return &model.KVPair{
+		Key:      v1key,
+		Value:    v1value,
+		Revision: kvp.Revision,
+	}, nil
+}
+
+func convertWorkloadEndpointV2ToV1Key(v2res *apiv2.WorkloadEndpoint) (model.WorkloadEndpointKey, error) {
+	parts := names.ExtractDashSeparatedParms(v2res.GetName(), 4)
+	if len(parts) != 4 {
+		return model.WorkloadEndpointKey{}, errors.New("Not enough information provided to create v1 Workload Endpoint Key")
+	}
+	return model.WorkloadEndpointKey{
+		Hostname:       parts[0],
+		OrchestratorID: parts[1],
+		WorkloadID:     v2res.GetNamespace() + "/" + parts[2],
```

Comment:
so namespace in the Key too - so you don't need the KVP Value to create the v1 Key.